process/economics/economicsData.go

```diff
@@ -1,41 +1,113 @@
 package economics
 
 import (
+	"math/big"
+	"strconv"
+
 	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/process"
 )
 
 // EconomicsData will store information about economics
 type EconomicsData struct {
-	rewardsValue        uint64
+	rewardsValue        *big.Int
 	communityPercentage float64
 	leaderPercentage    float64
 	burnPercentage      float64
-
-	minGasPrice      uint64
-	minGasLimitForTx uint64
-	minTxFee         uint64
-
-	communityAddress string
-	burnAddress      string
+	minGasPrice         uint64
+	minGasLimitForTx    uint64
+	communityAddress    string
+	burnAddress         string
 }
 
 // NewEconomicsData will create and object with information about economics parameters
-func NewEconomicsData(economics *config.ConfigEconomics) *EconomicsData {
+func NewEconomicsData(economics *config.ConfigEconomics) (*EconomicsData, error) {
+	//TODO check addresses what happens if addresses are wrong
+	rewardsValue, minGasPrice, minGasLimitForTx, err := convertValues(economics)
+	if err != nil {
+		return nil, err
+	}
+
+	notGreaterThanZero := rewardsValue.Cmp(big.NewInt(0))
+	if notGreaterThanZero < 0 {
+		return nil, process.ErrInvalidRewardsValue
+	}
+
+	err = checkValues(economics)
+	if err != nil {
+		return nil, err
+	}
+
 	return &EconomicsData{
-		rewardsValue:        economics.RewardsSettings.RewardsValue,
+		rewardsValue:        rewardsValue,
 		communityPercentage: economics.RewardsSettings.CommunityPercentage,
 		leaderPercentage:    economics.RewardsSettings.LeaderPercentage,
 		burnPercentage:      economics.RewardsSettings.BurnPercentage,
-		minGasPrice:         economics.FeeSettings.MinGasPrice,
-		minGasLimitForTx:    economics.FeeSettings.MinGasLimitForTx,
-		minTxFee:            economics.FeeSettings.MinTxFee,
+		minGasPrice:         minGasPrice,
+		minGasLimitForTx:    minGasLimitForTx,
```

Comment:
same naming problem here

Reply:
no need for ForTx. there will be only one min gas limit

Reply:
Done.